src/Lykke.Service.OperationsHistory/Controllers/OperationsHistoryController.cs

```diff
@@ -68,12 +69,34 @@ public OperationsHistoryController(IHistoryCache cache, IHistoryLogEntryReposito
                 return BadRequest(ErrorResponse.Create(ClientNotExists));
             }
 
-            return Ok(await _cache.GetAsync(clientId, operationType, assetId, take, skip));
+            var wallets = await _clientAccountService.GetWalletsByClientIdAsync(clientId);
+
+            var walletIds = wallets.Select(w => w.Type == nameof(WalletType.Trading) ? clientId : w.Id).Distinct();
+
+            var result = new List<IHistoryLogEntryEntity>();
+
+            await Task.WhenAll(
+                walletIds
+                    .Select(x => _cache.GetAsync(x, operationType, assetId)
```

Comment:
If _cache is behind a network and you have a lot of walletIds it potentially may open to many TCP connections